x-pack/plugins/security_solution/public/common/containers/source/index.tsx

```diff
@@ -155,19 +156,23 @@ export const useFetchIndex = (
           .subscribe({
             next: (response) => {
               if (isCompleteResponse(response)) {
-                const stringifyIndices = response.indicesExist.sort().join();
-
-                previousIndexesName.current = response.indicesExist;
-                setLoading(false);
-                setState({
-                  browserFields: getBrowserFields(stringifyIndices, response.indexFields),
-                  docValueFields: getDocValueFields(stringifyIndices, response.indexFields),
-                  indexes: response.indicesExist,
-                  indexExists: response.indicesExist.length > 0,
-                  indexPatterns: getIndexFields(stringifyIndices, response.indexFields),
+                Promise.resolve().then(() => {
+                  ReactDOM.unstable_batchedUpdates(() => {
+                    const stringifyIndices = response.indicesExist.sort().join();
+
+                    previousIndexesName.current = response.indicesExist;
+                    setLoading(false);
+                    setState({
+                      browserFields: getBrowserFields(stringifyIndices, response.indexFields),
+                      docValueFields: getDocValueFields(stringifyIndices, response.indexFields),
+                      indexes: response.indicesExist,
+                      indexExists: response.indicesExist.length > 0,
+                      indexPatterns: getIndexFields(stringifyIndices, response.indexFields),
```

Comment:
instead of these 3 functions we can call [`getDataViewStateFromIndexFields`](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/common/containers/source/use_data_view.tsx#L50-L87) which combines the 3 properties in one memoized reduce.